genie-agent/src/main/java/com/netflix/genie/agent/execution/services/JobExecutionEventService.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import com.netflix.genie.proto.JobCompletionEvent;
+import com.netflix.genie.proto.StateActionCompletionEvent;
+import com.netflix.genie.proto.StateChangeEvent;
+
+/**
+ * Service for publishing job execution events from various sub-systems.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public interface JobExecutionEventService {
+
+    /**
+     * Enqueue an state change event.
+     * @param event the event
+     */
+    void publishStateChange(final StateChangeEvent event);
```

Comment:
The interface is necessary to keep the implementation package-local.

I agree DTO rather than Protos in the public interface would be better.
On the other side, it's a lot of boilerplate to just convert a POJO into a different POJO, neither of which is ever exposed outside the agent anyway...
So just thought you'd let me get away with this :-P
I guess i'll do DTOs...